cnf-certification-test/lifecycle/podrecreation/podrecreation.go

```diff
@@ -140,11 +140,11 @@ func deletePod(pod *corev1.Pod, mode string, wg *sync.WaitGroup) error {
 	return nil
 }
 
-func CordonCleanup(node string) {
+func CordonCleanup(node string, check *checksdb.Check) {
 	err := CordonHelper(node, Uncordon)
 	if err != nil {
 		logrus.Errorf("cleanup: error uncordoning the node: %s, err=%s", node, err)
-		ginkgo.AbortSuite(fmt.Sprintf("cleanup: error uncordoning the node: %s, err=%s", node, err))
+		check.SetResultAborted(fmt.Sprintf("cleanup: error uncordoning the node: %s, err=%s", node, err))
```

Comment:
SetResultAborted does not seem to abort skip the subsequent tests. maybe we could propagate the aborted test info to the RunChecks function and use "OnAbort" to skip remaining tests, similar to the timeout and sigint

Reply:
In the latest commit, I added an `abortChan` to the `check` struct itself where the struct can abort itself if needed.  In Go channels are by reference by default similar to slices so we should be good creating the abort chan and passing it in to the check.